src/test/integrationTest/authRoute.test.js

```diff
@@ -0,0 +1,94 @@
+/* eslint-disable camelcase */
+import chai, { expect, should } from 'chai';
+import chaiHTTP from 'chai-http';
+import server from '../..';
+
+should();
+chai.use(chaiHTTP);
+const route = '/api/v1/auth/signup';
+const generateNumber = () => `${Math.round(Math.random() * 1000)}${Date.now()}`;
+const validUser = {
+  email: `john${generateNumber()}@doe.com`,
+  password: 'this1sN0TmyPassword'
+};
+
+const shortPassword = {
+  email: `john${generateNumber()}@doe.com`,
+  password: 'short'
+};
+
+const nonAlphaNumPassword = {
+  email: `john${generateNumber()}@doe.com`,
+  password: 'isThisInvalid!? :-)'
+};
+
+const invalidEmail = {
+  email: 'invalidEmail',
+  password: 'noNeedForPasswordAnyMore'
+};
+
+const nullEmail = {
+  password: 'noNeedForPasswordAnyMore'
+};
+
+const nullPassword = {
+  email: `john${generateNumber()}@doe.com`
+};
+
+const runTest = (done, object = {}) => {
+  chai
+    .request(server)
+    .post(route)
+    .send(object)
+    .then((res) => {
+      res.should.have.status(400);
+      res.body.should.be.an('object');
+      res.body.should.have.property('error');
+      const { status } = res.body;
+      expect(status).to.eql(res.status);
+      done();
+    })
+    .catch(done);
+};
+
+describe('USER VALIDATION TEST', () => {
+  before('Register a valid user', (done) => {
+    chai
```

Comment:
you should not be registering a valid user in your test. if your tests needs data to be in the database beforehand, then seed the database before running your test.

Reply:
I've resolved it. Thanks.